pkg/binary/envoy/debug/lsof.go

```diff
@@ -63,15 +61,7 @@ type openFilesDataCollection struct {
 }
 
 // retrieveOpenFilesData writes statistics of open files associated with envoy instance(s) to a json file
-// Errors from platform-specific libraries log to debug instead of raising an error or logging in an error category.
-// This avoids filling logs for unresolvable reasons.
 func (o *openFilesDataCollection) retrieveOpenFilesData() error {
-	f, err := os.Create(filepath.Join(o.lsofDir, "lsof.json"))
```

Comment:
re-organized to write file at the bottom, as doing so avoids writing an empty file